### PR TITLE
feat: support SHOW MASTER STATUS

### LINKED ISSUE #426: [SQL] Support `SHOW MASTER STATUS` SQL statement.
**What would you like to be added**:

**Why is this needed**:

https://dev.mysql.com/doc/refman/8.0/en/show-master-status.html

### PR DESCRIPTION
**Pull request by Jackson-soft**
**What this PR does**:

add support show master status

**Which issue(s) this PR fixes**:

Fixes #426

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/optimize/dal/show_master_status.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dal
+
+import (
+	"context"
```
Comment (dongzl) [inline pkg/runtime/optimize/dal/show_master_status.go:21]: Please format import, use this tool:

```
go get -u github.com/dubbogo/tools/cmd/imports-formatter

imports-formatter
```

File: pkg/runtime/plan/dal/show_master_status.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+package dal
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/arana-db/arana/pkg/proto"
+	"github.com/arana-db/arana/pkg/runtime/ast"
```
Comment (dongzl) [inline pkg/runtime/plan/dal/show_master_status.go:26]: The same to above.

**Review by dongzl — CHANGES_REQUESTED** [review]

File: pkg/runtime/plan/dal/show_master_status.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+func (s *ShowMasterStatusPlan) ExecIn(ctx context.Context, conn proto.VConn) (proto.Result, error) {
+	var (
+		sb   strings.Builder
+		args []int
+	)
+	ctx, span := plan.Tracer.Start(ctx, "ShowMasterStatusPlan.ExecIn")
+	defer span.End()
+
+	if err := s.stmt.Restore(ast.RestoreDefault, &sb, &args); err != nil {
+		return nil, fmt.Errorf("failed to execute SHOW MASTER STATUS statement, error: %w", err)
```
Comment (binbin0325) [inline pkg/runtime/plan/dal/show_master_status.go:56]: use perrors.Wrap might be better

**Review by binbin0325 — APPROVED** [review]

File: pkg/runtime/ast/show.go [hunk 4]
```diff
@@ -481,3 +482,17 @@ func (s *ShowCharset) Restore(flag RestoreFlag, sb *strings.Builder, args *[]int
 
 	return nil
 }
+
+type ShowMastStatus struct {
```
Comment (jjeffcaii) [inline pkg/runtime/ast/show.go:486]: Should be `ShowMasterStatus`.

File: pkg/runtime/plan/dal/show_master_status_test.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package dal
+
+import (
+	"github.com/arana-db/parser"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
```
Comment (jjeffcaii) [inline pkg/runtime/plan/dal/show_master_status_test.go:7]: Reformat imports.

**Review by jjeffcaii — APPROVED** [review]

File: pkg/runtime/plan/dal/show_master_status.go (hunk 3 above)
Comment (Jackson-soft) [reply pkg/runtime/plan/dal/show_master_status.go:56]: This is the official way

Comment (jjeffcaii) [reply pkg/runtime/plan/dal/show_master_status.go:58]: I think `perrors.Wrap` will be better, because it will pass the stack info.

**Review by dongzl — APPROVED** [review]

**Pull request merged**